src/overlays/actors/ovl_En_Mu/z_en_mu.c

```diff
@@ -44,27 +34,179 @@ static ColliderCylinderInit D_80AB0BD0 = {
+    func_80034F54(globalCtx, &this->jointTable, &this->morphTable, 0x10);
+}
+
+void EnMu_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnMu* this = THIS;
+    s32 pad;
+    f32 talkDist;
+    Vec3s pos;
+
+    pos.x = this->actor.world.pos.x;
+    pos.y = this->actor.world.pos.y;
+    pos.z = this->actor.world.pos.z;
+
+    this->collider.dim.pos = pos;
+
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+    SkelAnime_Update(&this->skelAnime);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    this->actionFunc(this, globalCtx);
+    talkDist = this->collider.dim.radius + 30.0f;
+    func_800343CC(globalCtx, &this->actor, this->dialogState, talkDist, EnMu_GetFaceReaction, EnMu_CheckDialogState);
+
+    this->actor.focus.pos = this->actor.world.pos;
+    this->actor.focus.pos.y += 60.0f;
+}
+
+s32 EnMu_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, s32 arg2, s32 arg3, Vec3s* rot, void* thisx) {
+    EnMu* this = THIS;
+    if ((limbIndex == 5) || (limbIndex == 6) || (limbIndex == 7) || (limbIndex == 0xB) || (limbIndex == 0xC) ||
+        (limbIndex == 0xD) || (limbIndex == 0xE)) {
+        rot->y += Math_SinS(this->jointTable[limbIndex]) * 200.0f;
+        rot->z += Math_CosS(this->morphTable[limbIndex]) * 200.0f;
+    }
+    return 0;
+}
+
+void EnMu_PostLimbDraw(s32 arg0, s32 arg1, s32 arg2, s32 arg3) {
+}
+
+Gfx* EnMu_DisplayListSetColor(GraphicsContext* gfxCtx, u8 r, u8 g, u8 b, u8 a) {
+    Gfx* dlist;
+    dlist = Graph_Alloc(gfxCtx, 2 * sizeof(Gfx));
+    gDPSetEnvColor(dlist, r, g, b, a);
+    gSPEndDisplayList(dlist + 1);
+    return dlist;
+}
+
+void EnMu_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnMu* this = THIS;
+
+    // 2 sets of 5 colors for each actor in town with different colors
+    Color_RGBA8 colors[2][5] = { 0x64, 0x82, 0xEB, 0x00, 0xA0, 0xFA, 0x3C, 0x00, 0x5A, 0x3C, 0x14, 0x00, 0x1E, 0xF0,
+                                 0xC8, 0x00, 0x8C, 0x46, 0x14, 0x00, 0x8C, 0x46, 0x14, 0x00, 0x1E, 0xF0, 0xC8, 0x00,
+                                 0x5A, 0x3C, 0x14, 0x00, 0xA0, 0xFA, 0x3C, 0x00, 0x64, 0x82, 0xEB, 0x00 }; //?
+    u8 D_80AB0C60[] = { 0x08, 0x09, 0x0A, 0x0B, 0x0C };                                                    //?
+    u32 i;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_mu.c", 514);
+    Matrix_Translate(-1200.0f, 0.0f, -1400.0f, 1);
+    for (i = 0; i != 5; i++) {
+        gSPSegment(POLY_OPA_DISP++, D_80AB0C60[i],
+                   EnMu_DisplayListSetColor(globalCtx->state.gfxCtx, colors[this->actor.params][i].r,
+                                            colors[this->actor.params][i].g, colors[this->actor.params][i].b,
+                                            colors[this->actor.params][i].a));
+    }
+    SkelAnime_DrawFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable,
+                          (s32)this->skelAnime.dListCount, EnMu_OverrideLimbDraw, EnMu_PostLimbDraw, this);
```

Comment:
```suggestion
                          this->skelAnime.dListCount, EnMu_OverrideLimbDraw, EnMu_PostLimbDraw, this);
```
this cast should be unneeded.